src/Core/Core/Execution/Middleware/QueryExecutionBuilderExtensions.cs

```diff
@@ -1,40 +1,63 @@
-﻿using System;
-using System.Collections.Generic;
+using System;
 using System.Linq;
-using System.Linq.Expressions;
-using System.Reflection;
+using HotChocolate.Execution.Configuration;
 using HotChocolate.Language;
 using HotChocolate.Runtime;
-using HotChocolate.Utilities;
 using HotChocolate.Validation;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace HotChocolate.Execution
 {
     public static class ClassQueryExecutionBuilderExtensions
     {
-        public static IQueryExecutionBuilder UseDiagnostics(
+        public static IQueryExecutionBuilder UseDefaultPipeline(
             this IQueryExecutionBuilder builder)
         {
-            return builder.Use<DiagnosticMiddleware>();
+            return builder
+                .UseDefaultPipeline(new QueryExecutionOptions());
         }
 
-        public static IQueryExecutionBuilder UseExceptionHandling(
+        public static IQueryExecutionBuilder UseDefaultPipeline(
+            this IQueryExecutionBuilder builder,
+            IQueryExecutionOptionsAccessor options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            return builder
+                .AddErrorHandler(options)
+                .AddQueryValidation(options)
+                .AddDefaultValidationRules()
+                .AddQueryCache(100)
```

Comment:
Should we offer here an accessor instead of `100`.

Reply:
Yes, I did forget this one.

Reply:
Ok, this one is now fixed.